tap_github/tests/test_tap.py

```diff
@@ -22,6 +26,27 @@ def test_validate_repo_list_config(repo_list_config):
     assert partitions == repo_list_context
 
 
+def alternative_sync_chidren(self, child_context: dict) -> None:
```

Comment:
Maybe worth adding a short docstring here?

Reply:
Also, I'm wondering about how stable that upstream method is. If its logic changes, we'll have to amend the code here as well to reflect that. Seeing how short it is, it's quite unlikely though...
As an alternative option, we could fiddle with the singer catalog at the start of the test run to deselect this stream if the conditions are not right.

Reply:
Added a docstring. I think it should be pretty stable and this gives us more control for future streams I think.